app/frontend/components/shared/permit-applications/requirement-form.tsx

```diff
@@ -10,13 +9,14 @@ interface IRequirementFormProps {
 }
 
 export function RequirementForm({ permitApplication, requirements }: IRequirementFormProps) {
-  const [form, setForm] = useState(permitApplication ? permitApplication.requirements : requirements)
-  // const [form, setForm] = useState(stubbedForm)
+  const form = permitApplication ? permitApplication.requirements : requirements
+  const [submission, setSubmission] = useState(permitApplication?.submissionData || {})
 
   //if there is no permit application provdided, you cannto run the key items like submit OR file upload.
   const onSubmit = (submission: any) => {
     if (permitApplication) {
       permitApplication.update({ submissionData: submission })
+      setSubmission(submission)
```

Comment:
What would happen if there is an error with the submission

Reply:
remvoed teh setSubmission.